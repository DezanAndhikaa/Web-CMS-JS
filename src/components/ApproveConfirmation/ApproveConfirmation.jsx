

import React from 'react';
import { DialogContent, Button, Modal } from '@material-ui/core';
import { Confirm } from '../../assets/imgs';
import './ApproveConfirmation.scss';
import CloseNotif from '../CloseNotif/CloseNotif';

export default class ApproveConfirmation extends React.PureComponent {
	render() {
		return (
			<Modal open={this.props.openModal} onClose={this.props.onClose} className="modal-container">
				<DialogContent className="confirmation-modal-content">
					<div className="confirmation-modal">
						<CloseNotif onClose={this.props.onClose}/>
						<div className="confirmation-container">
							<p className="confirmation-title">Approve Planning</p>
							<p className="confirmation-title">From Service Order</p>
							<img className="confirmation-image" src={Confirm} alt="" />
							<p className="confirmation-caption">Are you sure want to</p>
							<p className="confirmation-caption">approve {this.props.totalData} items?</p>
							<div className="btn-row">
								<Button className="button-no" onClick={this.props.onClose}>No</Button> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
								<Button className="button-yes" onClick={this.props.onApprove}>Yes</Button>
							</div>
						</div>
					</div>
				</DialogContent>
			</Modal>
		);
	}
}
