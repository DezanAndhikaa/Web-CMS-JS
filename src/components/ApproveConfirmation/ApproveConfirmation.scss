@import '../../themes/functions';
@import '../../themes/variables';

.modal-container-approve{
    width: fit-content;
    height: fit-content;
    margin: 12% 0 0 38%;
    .approve-confirmation-content{
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 0;
        .confirmation-modal-approve{
            width: vw(519);
            background-color: white;
            margin: auto;
            border-radius: vw(15);
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            padding: vw(15) vw(24) vw(30) vw(24);
            .confirmation-container-approve{
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;
                flex: 1;
                .confirmation-title-approve{
                    margin: 0;
                    padding-top: vw(20);
                    color: #3C3B3B;
                    font-family: 'Lato-Bold';
                    font-style: normal;
                    font-size: 24px;
                    line-height: 29px;
                }
                .confirmation-title-approve2{
                    margin: 0;
                    padding-top: vw(5);
                    color: #3C3B3B;
                    font-family: 'Lato-Bold';
                    font-style: normal;
                    font-size: 24px;
                    line-height: 29px;
                }
                .confirmation-image-approve{
                    width: vw(200);
                    margin-top: vw(16);
                    margin-bottom: vw(25);
                }
                .confirmation-caption-approve{
                    margin: 0;
                    font-family: 'Lato-Regular';
                    opacity: 0.51;
                    color: #3C3B3B;
                    font-weight: 500;
                    font-size: 13px;
                    line-height: 18px;
                }
            }
        }
    }
    .btn-row{
        flex: 1;
        display: flex;
        flex-direction: row;
        margin-top: vw(20);
        width: 100%;
        .button-rejected{
            display: flex;
            text-transform: capitalize;
            align-items: center;
            justify-content: center;
            width: 40%;
            height: vw(45.07);
            font-size: 12px;
            line-height: 14px;
            font-weight: bold;
            font-family: 'Lato-Regular';
            background-color: #F2F2F4;
            color: #1D1D1D;
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-rejected:hover {
            background-color: rgb(227, 227, 230);
        }
        .button-approved{
            display: flex;
            text-transform: capitalize;
            align-items: center;
            justify-content: center;
            width: 60%;
            height: vw(45.07);
            font-size: 12px;
            line-height: 14px;
            font-weight: bold;
            font-family: 'Lato-Regular';
            background-color: map-get($colors , primary);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            cursor: pointer;
        }
        .button-approved:hover {
            background-color: map-get($colors, primary-dark);
        }
    }
}

@media (min-width: 600px) and (max-width: 1025px) {    
    .modal-container-approve{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .approve-confirmation-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal-approve{
                width: vw-tab(450);
                height: vw-tab(300);
                background-color: white;
                margin: auto;
                border-radius: vw-tab(5);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-tab(15) vw-tab(24) vw-tab(30) vw-tab(24);
                .confirmation-container-approve{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title-approve{
                        margin: 0;
                        font-size: vw-tab(33);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image-approve{
                        width: vw-tab(150);
                        margin-top: vw-tab(16);
                        margin-bottom: vw-tab(25);
                    }
                    .confirmation-caption-approve{
                        margin: 0;
                        font-size: vw-tab(20);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}

@media (max-width: 599px) {    
    .modal-container-approve{
        width: fit-content;
        height: fit-content;
        margin: auto;
        .approve-confirmation-content{
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 0;
            .confirmation-modal-approve{
                margin: 0;
                width: vw-mobile(215);
                height: vw-mobile(200);
                background-color: white;
                border-radius: vw-mobile(15);
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: space-between;
                padding: vw-mobile(10);
                .confirmation-container-approve{
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    flex: 1;   
                    .confirmation-title-approve{
                        margin: 0;
                        font-size: vw-mobile(20);
                        font-family: 'SFUIDisplay-Bold';
                        color: #3C3B3B;
                    }
                    .confirmation-image-approve{
                        width: vw-mobile(100);
                        margin-top: vw-mobile(5);
                        margin-bottom: vw-mobile(10);
                    }
                    .confirmation-caption-approve{
                        margin: 0;
                        font-size: vw-mobile(15);
                        font-family: 'SFUIDisplay-Regular';
                        text-align: center;
                        opacity: 0.5;
                        color: #3C3B3B;
                    }
                }
            }
        }
    }
}