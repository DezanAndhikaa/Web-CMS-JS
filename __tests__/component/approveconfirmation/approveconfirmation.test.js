import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import ApproveConfirmation from "components/ApproveConfirmation/ApproveConfirmation";

Enzyme.configure({adapter:new Adapter()});
it("render modal approve confirmation correctly", () => {
  const wrapper = shallow (
    <ApproveConfirmation />
  )
  expect(wrapper).toMatchSnapshot();
})