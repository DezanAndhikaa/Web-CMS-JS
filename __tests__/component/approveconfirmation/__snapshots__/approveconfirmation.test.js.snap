exports[`render modal approve confirmation correctly 1`] = `
<ForwardRef(Modal)
  className="modal-container-approve"
>
  <WithStyles(ForwardRef(DialogContent))
    className="approve-confirmation-content"
  >
    <div
      className="confirmation-modal-approve"
    >
      <CloseNotif />
      <div
        className="confirmation-container-approve"
      >
        <p
          className="confirmation-title-approve"
        >
          Confirmation
        </p>
        <p
          className="confirmation-title-approve2"
        >
          Approved
        </p>
        <img
          alt=""
          className="confirmation-image-approve"
          src={null}
        />
        <p
          className="confirmation-caption-approve"
        >
          Are you sure want to Approve 
          <b>
             
             items
          </b>
          ?
        </p>
        <div
          className="btn-row"
        >
          <WithStyles(ForwardRef(Button))
            className="button-rejected"
          >
            No
          </WithStyles(ForwardRef(Button))>
                 
          <WithStyles(ForwardRef(Button))
            className="button-approved"
            onClick={[Function]}
          >
            Yes
          </WithStyles(ForwardRef(Button))>
        </div>
      </div>
    </div>
  </WithStyles(ForwardRef(DialogContent))>
</ForwardRef(Modal)>
`;